apps/common/behaviours/casework-submission.js

```diff
@@ -36,9 +36,9 @@ module.exports = conf => {
         const model = new Model(req.sessionModel.toJSON());
         req.log('info', `Sending icasework submission to ${model.url()}`);
         return model.save()
-          .then(data => {
-            req.log('info', `Successfully submitted case to icasework (${data.createcaseresponse.caseid})`);
-            req.sessionModel.set('caseid', data.createcaseresponse.caseid);
+          .then(response => {
+            req.log('info', `Successfully submitted case to icasework (${response.data.createcaseresponse.caseid})`);
+            req.sessionModel.set('caseid', response.data.createcaseresponse.caseid);
             client.increment('casework.submission.success');
```

Comment:
nice improvements

for the above you could do
```suggestion
await super.saveValues(req, res);
```
you could also wrap it in a try and catch and put an await inside instead of the `.then`
```
try {
          const response = await model.save();
          req.log('info', `Successfully submitted case to icasework (${response.data.createcaseresponse.caseid})`);
          req.sessionModel.set('caseid', response.data.createcaseresponse.caseid);
          client.increment('casework.submission.success');
          next();
        } catch (e) {
          req.log('error', `Casework submission failed: ${e.status}`);
          req.log('error', e.headers && e.headers['x-application-error-info']);
          client.increment('casework.submission.failed');
          next(new Error(e.body));
        }
```

Reply:
I have implemented this and also refactored the prepare function to async/await as well